livekit/src/room/data_stream/outgoing.rs

```
@@ -386,6 +386,52 @@ impl OutgoingStreamManager {
         Ok(info)
     }
 
+    /// Send bytes to participants in the room.
+    ///
+    /// This method sends an in-memory blob of bytes to participants in the room
+    /// as a byte stream. It opens a stream using the provided options, writes the
+    /// entire buffer, and closes the stream before returning.
+    ///
+    /// The `total_length` in the header is set from the provided data and is not
+    /// overridable by `options.total_length`.
+    pub async fn send_bytes(
+        &self,
+        data: impl AsRef<[u8]>,
+        options: StreamByteOptions,
+    ) -> StreamResult<ByteStreamInfo> {
```

I get why you wouldn't want to let a user override `total_length` but also I wonder if maybe this could be enforced at the type level instead? I bring this up because it seems like it could be a really easy thing to accidentally set (probably errantly) and not realize that it's not actually doing anything.

One idea - maybe you could add a `SendByteOptions` which doesn't have a `total_length` field in it but otherwise mirrors `StreamByteOptions`? I'd imagine if you went with this option, it might make sense at the same time to do the same thing with the already existing `send_text` too - ie, `options: StreamTextOptions` => `options: SendTextOptions` since it looks like that method also suffers from this same problem.

Alternatively if that's too burdensome maybe an assertion that checks to make sure it isn't set could be another approach, so it at least fails fast / panics versus just silently working.

---

I like this idea, but I think it should wait until v3. While I could do this for `send_bytes` now without breaking, I think it is better to avoid introducing a new type that would be inconsistent with `send_text`. For now, I’ve added a warning if this option is set for `send_bytes`.

For v3, something to consider is whether structs for defining options, not just those related to data streams, should have private fields. This would make API changes easier, and, for this case, it would also allow using the type state pattern to only allow setting options that make sense in the context. For example, `with_total_length()` could only be defined on `ByteStreamOptions<Incremental>` for example.